stdlib/format.ml

```diff
@@ -250,25 +253,38 @@ and pp_output_newline state = state.pp_out_newline ()
 and pp_output_spaces state n = state.pp_out_spaces n
 and pp_output_indent state n = state.pp_out_indent n
 
+(* Format a textual token *)
+let format_pp_text state size text =
+  state.pp_space_left <- state.pp_space_left - size;
+  pp_output_string state text;
+  if text <> "" then state.pp_is_new_line <- false
+
+(* Format a string by its length *)
+let format_string state s = format_pp_text state (String.length s) s
```

Comment:
With the possibility of users overriding the default `pp_out_string` function, there might be a difference between calling `pp_out_string` on an empty string vs not calling `pp_out_string` at all. Perhaps `format_pp_text` should only be called when `s <> ""`? Then the `text <> ""` test in `format_pp_text` could be removed, preserving the possibly unintentional current behavior of clearing `pp_is_new_line` when printing an empty string.